### ISSUE TITLE
SchemaRegistry.prototype.init() loads all the latest schemas into memory

### ISSUE THREAD

**Issue by codeburke** [waldophotos/kafka-avro#14]
**Problem**
Currently with `KafkaAvro.prototype.init()` is called it loads all the latest schemas in the Schema Registry into memory. This might be ok for smaller clusters, but it doesn't scale when you get into the 100's or 1000's of schemas in the Registry for several reasons:

1. The most common use case for any application is to write to a few topics, not the majority of them.
2. The Schema Registry service will be slammed on startup of an Node process, due to the REST API.
3. (Really 2a) Doing 1000s of network requests for all the subjects in the Registry will cause slow down on a normally very quick starting Node process.

For these reasons I'd like to suggest a change that the KafkaAvro constructor take an optional array of topics via the config that will be preloaded from the schema registry. If no array is passed then it defaults to the current load-all-the-schemas initialization.

I'm working on a PR to implement what I suggested, but wanted to raise the Issue now to see if there were any thoughts on this approach?

**Proposed Solution Example**
```
let kafkaAvro = new KafkaAvro({
    kafkaBrokers: ....,
    schemaRegistry: ....,
    topics: ['myTopicV1', 'myTopicV2']
});

// kafkaAvro.init() then kafkaAvro.getProducer()/.getConsumer(); ....

// There would also be an async method to allow loading topics from the SR after instantiation
kafkaAvro.loadTopics(['myTopicV3, ....']).then(() => {
    // do something with the new topics
});
```

**Comment by thanpolas** [comment]
Yup, makes sense, feel free to send a PR for that

**Comment by thanpolas** [comment]
fixed in 0.8.0
